# Data Analytics - Fouille de données

## Travail de groupe

In [3]:
import pandas as pd

urlPO = "http://localhost:8888/?token=c4d389a1a9e0a21726c636086275a9ce83afe1418d3667d7"

filePathPO = "C:/IESN/BAC3/QUADRIMESTRE 1/IG321 Business Intelligence/année 2023-2024/Data Analytics/travail 1 - fouille de données/Travail1BI/Data/pokemons.csv"
filePathArnaud = ""
filePathBenjamin = ""

file = pd.read_csv(filePathPO)

file.shape

(1017, 18)

In [4]:
file.head()

id        name      rank    generation evolves_from  type1   type2  hp  \
0   1   bulbasaur  ordinary  generation-i      nothing  grass  poison  45   
1   2     ivysaur  ordinary  generation-i    bulbasaur  grass  poison  60   
2   3    venusaur  ordinary  generation-i      ivysaur  grass  poison  80   
3   4  charmander  ordinary  generation-i      nothing   fire    None  39   
4   5  charmeleon  ordinary  generation-i   charmander   fire    None  58   

   atk  def  spatk  spdef  speed  total  height  weight  \
0   49   49     65     65     45    318       7      69   
1   62   63     80     80     60    405      10     130   
2   82   83    100    100     80    525      20    1000   
3   52   43     60     50     65    309       6      85   
4   64   58     80     65     80    405      11     190   

               abilities                                               desc  
0  overgrow chlorophyll   A strange seed was planted on its back at birt...  
1  overgrow chlorophyll   When the bulb on its back grows large, it appe...  
2  overgrow chlorophyll   The plant blooms when it is absorbing solar en...  
3     blaze solar-power   Obviously prefers hot places. When it rains, s...  
4     blaze solar-power   When it swings its burning tail, it elevates t...

## Description des données 
Les données contenues dans ce fichier décrivent chaque pokémon en partant de la génération 1 à la génération 9.
TODO : décrire concept de génération

### Lignes
Chaque ligne contient les données d'un pokémon en fonction de son identifiant dans le pokédex national.
Les id sont compris entre 1 et 1017.

### Colonnes
Chaque colonne contient les données du pokémon dont il est question. 

#### Description des colonnes 
* id - Identifiant dans le pokédex national (numéro unique correspondant au pokémon)
* name - Nom du pokémon
* rank - Niveau de rareté (ou est bébé)
* generation - Génération de première apparition
* evolves_from - Pré-évolution
* type1 - Premier type
* type2 - Second type
* hp - Nombre de points de vie
* atk - Nombre de points d'attaque physique
* def - Nombre de points de défense
* spatk - Nombre de points d'attaque spéciale
* spdef - Nombre de points de défense spéciale
* speed - Nombre de points de vitesse
* total - Somme des points d'attributs (hp, atk, def, ..., speed)
* height - Taille (en décimètres) 
* weight - Poids (en centaines de grammes)
* abilities - Talent(s) du pokémon
* desc - Description du pokémon dans le pokédex

In [17]:
pokemon = file

print(pokemon.groupby("type1")["hp"].mean())
print(pokemon.groupby("type2")["hp"].mean())

pokemonGrassAsType1 = pokemon.groupby("type1")
pokemonGrassAsType2 = pokemon.groupby("type2")

type1
bug         56.493976
dark        75.800000
dragon      79.594595
electric    63.741379
fairy       71.379310
fighting    76.075000
fire        69.292308
flying      70.000000
ghost       61.800000
grass       65.372549
ground      74.475000
ice         75.709677
normal      77.418803
poison      70.146341
psychic     71.783333
rock        67.929825
steel       70.588235
water       70.865672
Name: hp, dtype: float64
type2
None        67.532129
bug         57.333333
dark        77.166667
dragon      86.689655
electric    91.700000
fairy       63.800000
fighting    79.787879
fire        70.400000
flying      70.390000
ghost       69.655172
grass       67.666667
ground      80.000000
ice         88.058824
normal      67.769231
poison      61.414634
psychic     75.925000
rock        70.812500
steel       70.068966
water       61.100000
Name: hp, dtype: float64
